### PR TITLE
[api][SIMS #28]Feature/reassessment offering

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Move offering id from `sims.applications` to `sims.student_assessments`
- Created a new column  `sims.applications.current_assessment` to point to the active/current assessment.
- DB migrations to remove the legacy columns from `sims.applications` will be part of an upcoming PR.

### CODE REVIEW

**Review by ann-aot — APPROVED** [review]
LGTM 👍 just a minor comment

File: sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts [hunk 1]
```diff
@@ -97,6 +99,18 @@ export class ProgramInfoRequestController extends BaseController {
         "Program Information Request (PIR) not found.",
       );
     }
+    // Original assessment to be used as a reference.
+    // PIR process happens only during original assessment.
+    if (
+      application.currentAssessment.triggerType !==
+      AssessmentTriggerType.OriginalAssessment
+    ) {
+      throw new UnprocessableEntityException(
```
[the remark this answers is not in the document]
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:119]: Decorators added 😉

**Review by henryoforeh-aot — APPROVED** [review]

**Review by dheepak-aot — APPROVED** [review]
Good work @andrewsignori-aot  👍

**Review by guru-aot — APPROVED** [review]

**Review by guru-aot — APPROVED** [review]
LGTM, thanks for the explanation @andrewsignori-aot . As we said, we can discuss in dev sync up, if other devs have any questions on it.

**Pull request merged**